packages/cw-derive/tests/messages_generation.rs

```diff
@@ -0,0 +1,36 @@
+use cosmwasm_std::{Addr, Decimal, Response};
+
+pub struct Ctx;
+pub struct Error;
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, schemars::JsonSchema)]
+pub struct QueryResult;
+
+#[cw_derive::interface]
+pub trait Interface {
+    #[msg(exec)]
+    fn no_args_execution(&self, ctx: Ctx) -> Result<Response, Error>;
+
+    #[msg(exec)]
+    fn argumented_execution(&self, ctx: Ctx, addr: Addr, coef: Decimal, desc: String);
+
+    #[msg(query)]
+    fn no_args_query(&self, ctx: Ctx) -> Result<QueryResult, Error>;
+
+    #[msg(query)]
+    fn argumented_query(&self, ctx: Ctx, user: Addr) -> Result<QueryResult, Error>;
+}
+
+#[test]
+fn messages_constructible() {
```

Comment:
Nice test for compiling shape.

Another one to test that no other cases are generated would be:

```rust
fn parse_exec(e: ExecMsg) -> i64 {
  match e {
    ExecMsg:NoArgsExecution {} =>1,
    ExecMsg::ArgumentedExecution { .. } => 2,
  } 
}
```

Not sure if so useful, but another idea for a basic "does it compile" test to cover the codegen

Reply:
I can add it - not sure about usefullness, but wont hurt. But in followup (to avoid conflicts).